internal/impl/update.go

```diff
@@ -90,21 +91,50 @@ func (d *Devbox) updateDevboxPackage(
 	if err != nil {
 		return err
 	}
-	if existing != nil && existing.Version != newEntry.Version {
+	if existing == nil {
+		ux.Finfo(d.writer, "Resolved %s to %[1]s %[2]s\n", pkg, newEntry.Resolved)
+		d.lockfile.Packages[pkg.Raw] = newEntry
+		return nil
+	}
+
+	if existing.Version != newEntry.Version {
 		ux.Finfo(d.writer, "Updating %s %s -> %s\n", pkg, existing.Version, newEntry.Version)
 		if err := d.removePackagesFromProfile(ctx, []string{pkg.Raw}); err != nil {
 			// Warn but continue. TODO(landau): ensurePackagesAreInstalled should
 			// sync the profile so we don't need to do this manually.
 			ux.Fwarning(d.writer, "Failed to remove %s from profile: %s\n", pkg, err)
 		}
 		d.lockfile.Packages[pkg.Raw] = newEntry
-	} else if existing == nil {
-		ux.Finfo(d.writer, "Resolved %s to %[1]s %[2]s\n", pkg, newEntry.Resolved)
-		d.lockfile.Packages[pkg.Raw] = newEntry
-	} else {
-		ux.Finfo(d.writer, "Already up-to-date %s %s\n", pkg, existing.Version)
+		return nil
+	}
+
+	// Check if the package's system info is missing, or not complete.
+	if featureflag.RemoveNixpkgs.Enabled() {
```

Comment:
Most of the logic in this block could be encapsulated in lock file. Maybe:
```golang
if existing.CanBeUpdated(newEntry) {
  d.lockfile.Update(existing, newEntry)
}
``` 

or something like that. It could deal with version changes and also missing system stuff.

That way we could make `Package.Systems`  semi-private. (It still needs to be exported because it is part of the JSON)

Reply:
Also, is this logic block missing profile install/uninstall? e.g. if a package previously required nixpkgs and we replace it with one that does not require it, do we want to update the profile?

Reply:
yes, I agree this could likely be moved into the `lock` package. In general, I think quite a bit of the update code (beyond this block) could be better encapsulated. I may not have time to get to it though...

Reply:
> Also, is this logic block missing profile install/uninstall? e.g. if a package previously required nixpkgs and we replace it with one that does not require it, do we want to update the profile?

🤔 yeah, this one I am unsure about. In theory, the package's binary should be the exact same. 

Its also why I think it was okay and/or put a question mark after using `InputAddressedPath` in the `Package.Installable` [here](https://github.com/jetpack-io/devbox/blob/main/internal/devpkg/package.go#L157-L158).

@gcurtis would you have an opinion?